packages/world/src/NatureLib.sol

```diff
@@ -84,138 +89,127 @@ library NatureLib {
+      distribution[3] = 26; // 3 seeds: 26%
+
+      drops = new RandomDrop[](1);
+      drops[0] = RandomDrop(ObjectTypes.MelonSeed, distribution);
+      return drops;
+    }
 
-  function getLeafDrops(ObjectType objectType, uint256 randomSeed) internal view returns (ObjectAmount[] memory result) {
-    uint256 chance = TreeLib.getLeafDropChance(objectType);
-    uint256[] memory distribution = new uint256[](2);
-    distribution[0] = 100 - chance; // No sapling
-    distribution[1] = chance; // 1 sapling
-
-    // Get sapling options and their weights using distribution
-    (ObjectAmount[] memory saplingOptions, uint256[] memory weights) =
-      getDropWeights(objectType.getSapling(), distribution);
-
-    // Select sapling drop based on calculated weights
-    ObjectAmount memory saplingDrop = selectObjectByWeight(saplingOptions, weights, randomSeed);
-
-    if (saplingDrop.objectType != ObjectTypes.Null) {
-      result = new ObjectAmount[](2);
-      result[0] = ObjectAmount(objectType, 1);
-      result[1] = saplingDrop;
-    } else {
-      result = new ObjectAmount[](1);
-      result[0] = ObjectAmount(objectType, 1);
+    if (objectType == ObjectTypes.Pumpkin) {
+      // Expected return 1.86
+      uint256[] memory distribution = new uint256[](4);
+      distribution[0] = 10; // 0 seeds: 10%
+      distribution[1] = 20; // 1 seed:  20%
+      distribution[2] = 44; // 2 seeds: 44%
+      distribution[3] = 26; // 3 seeds: 26%
+
+      drops = new RandomDrop[](1);
+      drops[0] = RandomDrop(ObjectTypes.PumpkinSeed, distribution);
+      return drops;
     }
 
-    return result;
+    if (objectType.isLeaf()) {
+      uint256 chance = TreeLib.getLeafDropChance(objectType);
+      uint256[] memory distribution = new uint256[](2);
+      distribution[0] = 100 - chance; // No sapling
+      distribution[1] = chance; // 1 sapling
+
+      drops = new RandomDrop[](1);
+      drops[0] = RandomDrop(objectType.getSapling(), distribution);
+      return drops;
+    }
   }
 
   function getRandomOre(Vec3 coord) internal view returns (ObjectType) {
     uint256 randomSeed = getRandomSeed(coord);
 
     // Get ore options and their weights (based on remaining amounts)
-    ObjectAmount[] memory oreOptions = new ObjectAmount[](6);
-    uint256[] memory weights = new uint256[](6);
 
     ObjectType[7] memory oreTypes = ObjectTypeLib.getOreTypes();
+    uint256[] memory weights = new uint256[](oreTypes.length - 1);
+
     // Use remaining amounts directly as weights
     // Skip UnrevealedOre (index 0) since it's not a specific ore type
     for (uint256 i = 1; i < oreTypes.length; i++) {
-      oreOptions[i - 1] = ObjectAmount(oreTypes[i], 1);
       (, weights[i - 1]) = getCapAndRemaining(oreTypes[i]);
     }
 
     // Select ore based on availability
-    ObjectAmount memory selectedOre = selectObjectByWeight(oreOptions, weights, randomSeed);
+    return oreTypes[selectByWeight(weights, randomSeed) + 1];
```

Comment:
a +1 was added here because previously we were using oreOptions but now are reading from oreTypes directly, correct?

Reply:
Yeah